src/common/history.c

```diff
@@ -745,10 +746,8 @@ int dt_history_copy_and_paste_on_image(int32_t imgid, int32_t dest_imgid, gboole
   dt_mipmap_cache_remove(darktable.mipmap_cache, dest_imgid);
   dt_image_reset_final_size(imgid);
 
-  /* update the aspect ratio if the current sorting is based on aspect ratio, otherwise the aspect ratio will be
```

Comment:
IIRC I did that because the speed of copy/paste a in a large set of images was really slow. Computing the aspect ration need some CPU cycle. So the idea was to do that here only if the current sorting was based on aspect ratio (for obvious reasons) and delay that at some other point otherwise.

So, have you tested the perf?